### PR TITLE
zh-cn: init the translation of Firefox 123 Release note

### PR DESCRIPTION
**Pull request by yin1999**
### Description

init the translation of Firefox 123 Release note

### Related issues and pull requests

Part of: #17467

### CODE REVIEW

File: files/zh-cn/mozilla/firefox/releases/123/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: Firefox 123 的开发者说明
+slug: Mozilla/Firefox/Releases/123
+l10n:
+  sourceCommit: 1822cdf5a86574429c4c49883a402663ef16a4ef
```
Comment (yin1999) [inline files/zh-cn/mozilla/firefox/releases/123/index.md:5]:
```suggestion
  sourceCommit: a9bd012040da7024cc05c00336aa7ebb42c5c227
```

File: files/zh-cn/mozilla/firefox/releases/123/index.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+本文提供了有关 Firefox 123 中影响开发者的变更信息。Firefox 123 于 [2024 年 2 月 20 日](https://whattrainisitnow.com/release/?version=122)发布。
+
+## 为 Web 开发者带来的变化
+
+### 开发者工具
+
+### HTML
+
+- {{htmlelement("template")}} 元素现已支持 `shadowrootmode` 属性，允许声明式创建影子 DOM 子树。该属性可以设置为 `open` 或 `closed`，分别对外部代码中公开或隐藏影子 DOM 中的 JavaScript。其可设置的值与 `Element.attachShadow()` 方法的 `mode` 选项相同。（[Firefox bug 1870052](https://bugzil.la/1870052)）
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/firefox/releases/123/index.md:18]:
```suggestion
- {{htmlelement("template")}} 元素现已支持 `shadowrootmode` 属性，允许声明式创建影子 DOM 子树。该属性可以设置为 `open` 或 `closed`，分别对外部代码公开或隐藏影子 DOM 中的 JavaScript。其可设置的值与 `Element.attachShadow()` 方法的 `mode` 选项相同。（[Firefox bug 1870052](https://bugzil.la/1870052)）
```

File: files/zh-cn/mozilla/firefox/releases/123/index.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+### JavaScript
+
+- 修复了 {{jsxref("Date.parse()")}} 全局对象的一系列错误，使其与其他浏览器解析传递的值的方式保持一致。
+  - 月份错误的日期（例如“31 April”）现在会跳到下一个月（例如“1 May”）。（[Firefox bug 1872333](https://bugzil.la/1872333)）。
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/firefox/releases/123/index.md:23]: 原句读起来像“不合法的月份”

```suggestion
  - 月份中错误的日期（例如“31 April”）现在会跳到下一个月（例如“1 May”）。（[Firefox bug 1872333](https://bugzil.la/1872333)）。
```

**Review by jasonren0403 — COMMENTED** [review]
一些建议

**Review by jasonren0403 — APPROVED** [review]
LGTM 👍

**Pull request merged**